### PR TITLE
Fix Babelfish log function incorrect return result

### PR DESCRIPTION
**Pull request by RIC06X**
### Description

Previously Babelfish natural log function mapped to postgres base 10 log function and returned incorrect result. In addition,  Babelfish's two argument log function had base as first argument and value as second argument which is different from TSQL. This commit remapped Babelfish log() function to postgres ln() function and fixed the incorrect log value, and fixed the argument ordering issue for the two argument log function.

### Issues Resolved

BABEL-730

### Test Scenarios Covered ###
* return type test for log(x), log(x, y), log10(x)
* return value test for log(x), log(x, y), log10(x)
* function/procedure usage test for log(x), log(x, y), log10(x)
* view usage test for log(x), log(x, y), log10(x)
* implicit casting (string value) test for log(x), log(x, y), log10(x)

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by forestkeeper — DISMISSED** [review]

File: test/JDBC/input/BABEL-730-vu-cleanup.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
+use test_babel730
+go
```
Comment (forestkeeper) [inline test/JDBC/input/BABEL-730-vu-cleanup.sql]: We should add this to mvu schedule files

Comment (RIC06X) [reply test/JDBC/input/BABEL-730-vu-cleanup.sql]: Solved

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.4.0--3.5.0.sql [hunk 2]
```diff
@@ -37,6 +37,17 @@ LANGUAGE plpgsql;
...
+RETURNS FLOAT  AS 'babelfishpg_tsql','numeric_log_natural' LANGUAGE C STRICT IMMUTABLE PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION sys.bbf_log(FLOAT) TO PUBLIC;
+
+CREATE OR REPLACE FUNCTION sys.bbf_log(IN arg1 FLOAT, IN arg2 INT)
+RETURNS FLOAT  AS 'babelfishpg_tsql','numeric_log_base' LANGUAGE C STRICT IMMUTABLE PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION sys.bbf_log(FLOAT, INT) TO PUBLIC;
+
+CREATE OR REPLACE FUNCTION sys.bbf_log10(IN arg1 FLOAT)
+RETURNS FLOAT  AS 'babelfishpg_tsql','numeric_log10' LANGUAGE C STRICT IMMUTABLE PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION sys.bbf_log10(FLOAT) TO PUBLIC;
```
Comment (xhfanhe) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.4.0--3.5.0.sql]: Is public access necessary?

File: test/JDBC/expected/BABEL-730-vu-cleanup.out [hunk 3]
```diff
@@ -0,0 +1,21 @@
+-- drop table/view/function/procedure
+drop table test_log
```
Comment (xhfanhe) [inline test/JDBC/expected/BABEL-730-vu-cleanup.out:2]: Where is this table created?

File: test/python/expected/upgrade_validation/expected_dependency.out [hunk 4]
```diff
@@ -224,6 +224,9 @@ Function sys.bbf_binary_varbinary_cmp(sys.bbf_binary,sys.bbf_varbinary)
 Function sys.bbf_get_context_info()
 Function sys.bbf_get_current_physical_schema_name(text)
 Function sys.bbf_is_shared_schema(text)
+Function sys.bbf_log(double precision)
```
Comment (xhfanhe) [inline test/python/expected/upgrade_validation/expected_dependency.out]: Please add upgrade tests for bbf_log functions as well (if it's public accessible)

File: test/JDBC/expected/BABEL-730-vu-cleanup.out (hunk 3 above)
Comment (RIC06X) [reply test/JDBC/expected/BABEL-730-vu-cleanup.out:2]: this table is created in verify.sql, it is created by 'select into' stmt

File: test/python/expected/upgrade_validation/expected_dependency.out (hunk 4 above)
Comment (RIC06X) [reply test/python/expected/upgrade_validation/expected_dependency.out]: Tests for bbf_log functions are at the end of verify.sql

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.4.0--3.5.0.sql (hunk 2 above)
Comment (RIC06X) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.4.0--3.5.0.sql]: no, it is not necessary to have public access, I'll remove the grant to public stmt.

**Review by xhfanhe — APPROVED** [review]

**Review by forestkeeper — APPROVED** [review]

**Pull request merged**